garage/tf/policies/categorical_conv_policy.py

```diff
@@ -42,44 +41,47 @@ def __init__(
-            prob_network = ConvNetwork(
-                input_shape=env_spec.observation_space.shape,
-                output_dim=env_spec.action_space.n,
-                conv_filters=conv_filters,
-                conv_filter_sizes=conv_filter_sizes,
-                conv_strides=conv_strides,
-                conv_pads=conv_pads,
-                hidden_sizes=hidden_sizes,
-                hidden_nonlinearity=hidden_nonlinearity,
-                output_nonlinearity=output_nonlinearity,
-                name="prob_network",
-            )
-
-        self._l_prob = prob_network.output_layer
-        self._l_obs = prob_network.input_layer
-        self._f_prob = tensor_utils.compile_function(
-            [prob_network.input_layer.input_var],
-            L.get_output(prob_network.output_layer))
-
-        self._dist = Categorical(env_spec.action_space.n)
-
-        super(CategoricalConvPolicy, self).__init__(env_spec)
-        LayersPowered.__init__(self, [prob_network.output_layer])
+        self._prob_network_name = "prob_network"
+
+        with tf.variable_scope(name, "CategoricalConvPolicy"):
+            if prob_network is None:
+                prob_network = ConvNetwork(
+                    input_shape=env_spec.observation_space.shape,
+                    output_dim=env_spec.action_space.n,
+                    conv_filters=conv_filters,
+                    conv_filter_sizes=conv_filter_sizes,
+                    conv_strides=conv_strides,
+                    conv_pads=conv_pads,
+                    hidden_sizes=hidden_sizes,
+                    hidden_nonlinearity=hidden_nonlinearity,
+                    output_nonlinearity=output_nonlinearity,
+                    name="conv_prob_network",
+                )
+
+            with tf.name_scope(self._prob_network_name):
+                out_prob = L.get_output(prob_network.output_layer)
+
+            self._l_prob = prob_network.output_layer
+            self._l_obs = prob_network.input_layer
+            self._f_prob = tensor_utils.compile_function(
+                [prob_network.input_layer.input_var], [out_prob])
+
+            self._dist = Categorical(env_spec.action_space.n)
+
+            super(CategoricalConvPolicy, self).__init__(env_spec)
+            LayersPowered.__init__(self, [prob_network.output_layer])
 
     @property
     def vectorized(self):
         return True
 
     @overrides
-    def dist_info_sym(self,
-                      obs_var,
-                      state_info_vars=None,
-                      name="dist_info_sym"):
-        with enclosing_scope(self._name, name):
-            return dict(
-                prob=L.get_output(self._l_prob,
-                                  {self._l_obs: tf.cast(obs_var, tf.float32)}))
+    def dist_info_sym(self, obs_var, state_info_vars=None, name=None):
+        with tf.name_scope(name, "dist_info_sym", [obs_var]):
+            with tf.name_scope(self._prob_network_name, [obs_var]):
```

Comment:
what is the purpose of this second name_scope

Reply:
When the function [get_output](https://github.com/rlworkgroup/garage/blob/efb8e142206754582e4c83ec130024020e16a21e/garage/tf/core/layers.py#L2065) is called, a new instance of the operations in the neural network defined by the output layers provided as input is generated. Before calling this function it's necessary to add a scope to enclose all the operations that are generated, which is something that I did for the code on the TF branch, so it's clear in the graph that those operations belong to the networks.
Althogh this two scopes may seem redundant, since there can be more than one instance of the nerual network operations, add the scope for dist_info_sym makes more clear where the instances produced by ths method are in the graph.